### PR TITLE
[CARBONDATA-1113] Added validation for partition columns

### PR DESCRIPTION
**Pull request by manishgupta88**
Add various validation to partition columns like:-
1. Partition columns cannot be drop using alter command
2. Range partition info should have values in ascending order

### CODE REVIEW

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala [hunk 1]
```diff
@@ -169,7 +169,7 @@ object CommonUtil {
       partitionType.get.toUpperCase() match {
         case "HASH" => if (!numPartitions.isDefined) isValid = false
         case "LIST" => if (!listInfo.isDefined) isValid = false
-        case "RANGE" => if (!rangeInfo.isDefined) isValid = false
+        case "RANGE" => isValid = validateRangeInfo(rangeInfo)
```
Comment (lionelcao) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala]: Hi @manishgupta88, the rangeInfo validation should base on column data type, actually the validate function has been developed in my dev branch. Will submit in my next pr. Thanks although. ：）

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2152/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/17/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2239/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2241/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/112/

**Comment by asfgit** [comment]
Tests Passed for Spark1.6

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/114/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**